types/youtube-dl/index.d.ts

```diff
@@ -0,0 +1,23 @@
+// Type definitions for youtube-dl 1.12
+// Project: https://www.npmjs.com/package/youtube-dl
+// Definitions by: Bogdan Surai <https://github.com/bsurai>
+// Definitions: https://github.com/DefinitelyTyped/DefinitelyTyped
+// TypeScript Version: 2.4
+
+/// <reference types="node" />
+import * as fs from "fs";
+
+interface Info {
+    _filename: string;
+    filename: string;
+    size: number;
+}
+
+interface Youtubedl {
```

Comment:
Now that I look at this, I think it can be:
```ts
export = youtubedl;
declare function youtubedl(url: string, arg: string[], opt: {[key: string]: string}): youtubedl.Youtubedl;
declare namespace youtubedl {
    interface Youtubedl { ... }
    interface Info { ... }
}
```

Reply:
@andy-ms Done. May I ask you? Why do you think this approach is prefer?

Reply:
The only way to expose types from an `export = `module is to merge a namespace with the thing that's exported. So you can now do:
```ts
import youtubedl = require("youtubedl");
const info: youtubedl.Info = ...
```